pandas/core/resample.py

```diff
@@ -1025,6 +1025,9 @@ class TimeGrouper(Grouper):
     Use begin, end, nperiods to generate intervals that cannot be derived
     directly from the associated object
     """
+    _attributes = ('key', 'level', 'freq', 'sort', 'closed', 'label',
+                   'how', 'nperiods', 'axis', 'fill_method', 'limit',
```

Comment:
only should have public attributes

Reply:
iow you don’t need any special ones for TimeGrouper (which is publicly deprecated anyhow)

Reply:
A ``TimeGrouper`` object is returned when calling ``Grouper `` with a ``freq`` arg, IOW it's only the top level location of ``TimeGrouper`` that is deprecated, right?

Reply:
yes however you will never actually directly construct one
point being is the args are the same as Grouper (maybe a couple of extra which are prob not documented)

Reply:
can u add the missing arguments in the doc string of Grouper? eg  convention, base, loffset i think are only public ones

Reply:
Ah of course. Changed.

Reply:
Considering your last comment, it's actually possible to to:

```python
>>> pd.Grouper(key='A', freq='H', convention='s')
TimeGrouper(key='A', freq=<Hour>, axis=0, sort=True)
```

Note the output now misses ``convention='s'``. shouldn't ``TimeGrouper`` actually still have its own ``_attributes`` to account for this?

Reply:
yeah it should take a copy of the super (Grouper) and add to it

Reply:
Added ``_attributes``.

I don't know what ``base``, ``loffset`` and ``convention`` do, and can't quite figure it out from the code, so can't add explanations to the doc string, sorry. I added them, but without explanations. Do you have a text for each?